documentation/api/ext/v1/resource-graphs.markdown

```diff
@@ -0,0 +1,151 @@
+---
+title: "PuppetDB 3.2 » Extensions API (PE only)"
+layout: default
+canonical: "/puppetdb/latest/api/ext/v1/resource-graphs.html"
+---
+
+[paging]: ../../query/v4/paging.html
+[query]: ../../query/v4/query.html
+[subqueries]: ../../query/v4/operators.html#subquery-operators
+[environments]: ../../query/v4/environments.html
+[nodes]: ../../query/v4/nodes.html
+[statuses]: /puppet/latest/reference/format_report.html#puppettransactionreport
+
+You can query resource-graphs by making an HTTP request to the
+`/pdb/ext/v1/resource-graphs` endpoint.
+
+## `/pdb/ext/v1/resource-graphs`
+
+This will return a JSON array containing all the resource-graphs for each node
+in your infrastructure.
+
+This resource-graph is a report and a catalog munged together so that all
+resource information from a puppet run (parameters from catalogs, events from
+reports, etc.) is present in one place.
+
+### URL Parameters
+
+* `query`: Optional. A JSON array containing the query in prefix notation
+  (`["<OPERATOR>", "<FIELD>", "<VALUE>"]`). See the sections below for the
+  supported operators and fields. For general info about queries, see
+  [the page on query structure.][query]
+
+If a query parameter is not provided, all results will be returned.
+
+### Query Operators
+
+See [the Operators page.](../../query/v4/operators.html)
+
+### Query Fields
+
+* `certname` (string): the certname associated with the resource-graph
+* `environment` (string): the environment assigned to the node that submitted
+  the report.
+* `transaction_uuid` (string): string used to identify a puppet run.
+* `catalog_uuid` (string): a string used to tie a catalog to its associated
+  reports
+* `code_id` (string): a string used to tie a catalog to the Puppet code which
```

Comment:
I think we need (maybe not here) more detailed information about this and the catalog_uuid. Maybe we can link to their definitions in the Puppet docs? @joshcooper might be able to point us in the right direction.